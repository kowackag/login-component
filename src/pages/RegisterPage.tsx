import { FieldErrors, UseFormRegister, useForm } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";

import { RegisterPageTemplate } from "@/components/templates/RegisterPageTemplate";

import { registerShema } from "./authSchema";
import { FormDataType } from "@/utils/shared.types";

export type RegisterFieldsTypes = {
	email: string;
	password: string;
};
export type RegisterFormType = {
	onSubmit: React.FormEventHandler;
	errors?: FieldErrors<RegisterFieldsTypes>;
	register: UseFormRegister<RegisterFieldsTypes>;
};

type RegisterFormDataTypes = {
	email: string;
	password: string;
};

function RegisterPage() {
	const {
		register,
		formState: { errors },
		handleSubmit,
	} = useForm<RegisterFormDataTypes>({
		resolver: zodResolver(registerShema),
		defaultValues: {
			email: "",
			password: "",
		},
	});
	const onSubmit = handleSubmit((data: FormDataType) => console.log(data));
	return <RegisterPageTemplate onSubmit={onSubmit} register={register} errors={errors} />;
}

export default RegisterPage;
