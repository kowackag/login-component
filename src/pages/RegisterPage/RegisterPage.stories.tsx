import { Meta, StoryObj } from "@storybook/react";
import  RegisterPage  from "./RegisterPage";

const meta: Meta<typeof RegisterPage> = {
	component: RegisterPage,
	render: () => {
		
		return <RegisterPage  />;
	},
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
	args: {},
};
