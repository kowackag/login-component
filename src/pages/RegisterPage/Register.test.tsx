import { expect, it, describe } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import RegisterPage from "./RegisterPage";
import { BrowserRouter } from "react-router-dom";

describe("RegisterPage", () => {
	it("component exists", () => {
		const wrapper = render(
			<BrowserRouter>
				<RegisterPage />
			</BrowserRouter>,
		);
		expect(wrapper).toBeTruthy();
		expect(wrapper.container).toBeInTheDocument();
	});

	it("render appropriate title and form", () => {
		render(
			<BrowserRouter>
				<RegisterPage />
			</BrowserRouter>,
		);
		const form = screen.getByTestId("register-form");
		expect(form).toBeInTheDocument();
		const title = screen.getByRole("heading", { name: "Sign up" });
		expect(title).toBeInTheDocument();
	});
	it("renders correct errors when email/password does not meet requirements", async () => {
		render(
			<BrowserRouter>
				<RegisterPage />
			</BrowserRouter>,
		);
		const button = screen.getByRole("button");
		expect(button).toHaveTextContent("sign up");

		fireEvent.click(button);

		const errorList = await screen.findAllByTestId("error");
		expect(errorList).toHaveLength(2);
		expect(errorList[0]).toHaveTextContent("Invalid email address");
		expect(errorList[1]).toHaveTextContent("Invalid password");
	});
	it("hides error after filling form correctly", async () => {
		const { container } = render(
			<BrowserRouter>
				<RegisterPage />
			</BrowserRouter>,
		);
		const button = screen.getByRole("button");
		expect(button).toHaveTextContent("sign up");

		fireEvent.click(button);

		const errorList = await screen.findAllByTestId("error");
		expect(errorList).toHaveLength(2);

		const passwordInput = container.querySelector("#password") as HTMLInputElement;

		fireEvent.change(passwordInput, { target: { value: "own-password" } });
		expect(passwordInput.value).toBe("own-password");

		fireEvent.click(button);
		const newErrorList = await screen.findAllByTestId("error");
		// ??????
		console.log(newErrorList);
		expect(newErrorList).toHaveLength(2);
	});

	it("is reset after processing form", async () => {
		const { container } = render(
			<BrowserRouter>
				<RegisterPage />
			</BrowserRouter>,
		);
		const button = screen.getByRole("button");
		expect(button).toHaveTextContent("sign up");

		const passwordInput = container.querySelector("#password") as HTMLInputElement;

		fireEvent.change(passwordInput, { target: { value: "own-password" } });

		const emailInput = container.querySelector("#email") as HTMLInputElement;
		fireEvent.change(emailInput, { target: { value: "someone@wp.pl" } });

		expect(passwordInput.value).toBe("own-password");
		expect(emailInput.value).toBe("someone@wp.pl");

		fireEvent.click(button);

		expect(passwordInput.innerHTML).toBe("");
		expect(emailInput.innerHTML).toBe("");
	});

	it("renders correctly", () => {
		const { container } = render(
			<BrowserRouter>
				<RegisterPage />
			</BrowserRouter>,
		);
		expect(container).toMatchSnapshot();
	});
});
