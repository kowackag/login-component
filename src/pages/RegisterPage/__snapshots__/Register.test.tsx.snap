// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RegisterPage > renders correctly 1`] = `
<div>
  <div
    class="flex min-h-screen w-full bg-navy-blue"
  >
    <section
      class="m-auto min-h-[556px] w-4/5 rounded-lg bg-white px-11 py-10 font-primaryRegular shadow-md-2 sm:w-96"
    >
      <h1
        class="mb-4 text-lg font-bold uppercase"
        data-testid="title"
      >
        Sign up
      </h1>
      <form
        data-testid="register-form"
      >
        <div
          class="relative py-3"
        >
          <label
            class="inline-block font-primaryMedium"
            for="email"
          >
            Email
          </label>
          <input
            class="w-full rounded-md border-2 border-neutral-300 bg-transparent px-2 leading-9 outline-none"
            data-testid="email-input"
            id="email"
            name="email"
            type="text"
          />
        </div>
        <div
          class="relative py-3"
        >
          <label
            class="inline-block font-primaryMedium"
            for="password"
          >
            Password
          </label>
          <input
            class="w-full rounded-md border-2 border-neutral-300 bg-transparent px-2 leading-9 outline-none"
            data-testid="password-input"
            id="password"
            name="password"
            type="password"
          />
        </div>
        <button
          class="my-2 w-full rounded-lg bg-bright-rose text-center font-primaryMedium text-sm uppercase leading-10 tracking-widest text-neutral-100 shadow-btn"
          data-testid="btn-submit"
          type="submit"
        >
          sign up
        </button>
      </form>
      <div
        class="my-3 flex items-center"
      >
        <span
          class="flex-grow border border-b border-b-gray-300"
          data-testid="hrLine"
        />
        <p
          class="rounded-md border-2  border-neutral-300 px-1 py-0.5 font-primaryRegular uppercase text-neutral-400"
        >
          or
        </p>
        <span
          class="flex-grow border border-b border-b-gray-300"
          data-testid="hrLine"
        />
      </div>
      <div
        class="flex justify-center"
        data-testid="icon-box"
      >
        <a
          aria-label="google"
          class="mx-2 my-2"
          data-testid="link"
          href="https://mail.google.com/"
          target="_blank"
        >
          <svg
            height="36px"
            width="36px"
          >
            <use
              href="/google.svg#google"
            />
          </svg>
        </a>
        <a
          aria-label="facebook"
          class="mx-2 my-2"
          data-testid="link"
          href="https://www.facebook.com/"
          target="_blank"
        >
          <svg
            height="36px"
            width="36px"
          >
            <use
              href="/facebook.svg#facebook"
            />
          </svg>
        </a>
        <a
          aria-label="linkedin"
          class="mx-2 my-2"
          data-testid="link"
          href="https://www.linkedin.com/in/malgorzata-kowacka/"
          target="_blank"
        >
          <svg
            height="36px"
            width="36px"
          >
            <use
              href="/linkedin.svg#linkedin"
            />
          </svg>
        </a>
      </div>
      <div
        class="mb-0 mt-4 flex justify-center font-primaryMedium"
      >
        <p>
          Already a user?
        </p>
        <button
          class="ml-1 uppercase underline"
        >
          login
        </button>
      </div>
    </section>
  </div>
</div>
`;
